app/modules/api/tests/reducer.test.js

```diff
@@ -0,0 +1,64 @@
+// @flow
+
+import reducer from '../reducer';
+import * as t from '../actionTypes';
+import { statusTypes } from '../model';
+
+describe(`reducer`, (): void => {
+
+  it(`returns the initial state, when state parameter is undefined`, (): void => {
+    const dummyAction = {
+      type: t.SET_STATUS_ERROR,
+      error: {
+        message: `Flow will complain if the passed action isn't some kind of valid ApiAction.`,
+      },
+    };
+
+    expect(reducer(undefined, dummyAction)).toEqual({});
+  });
+
+  it(`handles api SET_STATUS action on empty state`, (): void => {
+    const prevState = {};
+
+    const setStatusAction = {
+      type: t.SET_STATUS,
+      payload: {
+        request: 'foobar',
+        status: statusTypes.SUCCESS,
+      },
+    };
+
+    const nextState = {
+      foobar: {
+        status: statusTypes.SUCCESS,
+      },
+    };
+
+    expect(reducer(prevState, setStatusAction)).toEqual(nextState);
```

Comment:
In reducer tests, please also test that the reducer does not accidentally mutate the state:
```
const resultState = reducer(prevState, setStatusAction);
expect(resultState).toEqual(next.state);
expect(resultState).not.toBe(prevState);
[...*]
```
\* If resultState contains objects that had their values changed, also test if these are not equal by reference. See `modules/content-items/tests/reducer.test.js` for examples.